node/dataSources/catalog.ts

```diff
@@ -136,21 +164,66 @@ export class CatalogDataSource extends IODataSource {
     return this.http.getRaw<T>(`/proxy/catalog${url}`, config)
   }
 
+  private collectionsUrl = (query: string) =>
+    `/pvt/collection/search/${query}?pageSize=50`
+
   private productSearchUrl = ({
     query = '',
     category = '',
-    specificationFilters,
+    specificationFilters = [],
     priceRange = '',
     collection = '',
     salesChannel = '',
     orderBy = '',
     from = 0,
     to = 9,
-    map = ''
+    map = '',
   }: ProductsArgs) => {
+    const BASE_URL = '/pub/products/search/'
+
     const sanitizedQuery = encodeURIComponent(decodeURIComponent(query).trim())
-    return (
-      `/pub/products/search/${sanitizedQuery}?${category && !query && `&fq=C:/${category}/`}${(specificationFilters && specificationFilters.length > 0 && specificationFilters.map(filter => `&fq=${filter}`)) || ''}${priceRange && `&fq=P:[${priceRange}]`}${collection && `&fq=productClusterIds:${collection}`}${salesChannel && `&fq=isAvailablePerSalesChannel_${salesChannel}:1`}${orderBy && `&O=${orderBy}`}${map && `&map=${map}`}${from > -1 && `&_from=${from}`}${to > -1 && `&_to=${to}`}`
+
+    let queryString = '?'
```

Comment:
You can use the querystring util to build the string.: https://www.npmjs.com/package/query-string

Reply:
I considered using it, but all the keys in the query string are `fq`, so using the same logic with an object wouldn't work.